lib/handlers/wsHandler.js

```diff
@@ -0,0 +1,236 @@
+'use strict';
+
+const co = require('co');
+const WebSocket = require('ws');
+const logUtil = require('../log');
+
+/**
+ * construct the request headers based on original connection,
+ * but delete the `sec-websocket-*` headers as they are already consumed by AnyProxy
+ */
+function getNoWsHeaders(headers) {
+  const originHeaders = Object.assign({}, headers);
+
+  Object.keys(originHeaders).forEach((key) => {
+    // if the key matchs 'sec-websocket', delete it
+    if (/sec-websocket/ig.test(key)) {
+      delete originHeaders[key];
+    }
+  });
+
+  delete originHeaders.connection;
+  delete originHeaders.upgrade;
+  return originHeaders;
+}
+
+/**
+ * get request info from the ws client
+ * @param @required wsClient the ws client of WebSocket
+ */
+function getWsReqInfo(wsReq) {
```

Comment:
I noticed that some weboscet url has not `:{port}`.
So, you need to normalize `port` before building `url`.

```js
/**
 * get request info from the ws client
 * @param @required wsClient the ws client of WebSocket
 */
function getWsReqInfo(wsReq) {
  const headers = wsReq.headers || {};
  const host = headers.host;
  const hostname = host.split(':')[0];
  const port = host.split(':')[1];
  // https://github.com/alibaba/anyproxy/pull/450
  const portPart = port ? `:${port}` : '';
  // TODO 如果是windows机器，url是不是全路径？需要对其过滤，取出
  const path = wsReq.url || '/';
  const isEncript = wsReq.connection && wsReq.connection.encrypted;

  return {
    url: `${isEncript ? 'wss' : 'ws'}://${hostname}${portPart}${path}`,
    headers: headers, // the full headers of origin ws connection
    noWsHeaders: getNoWsHeaders(headers),
    secure: Boolean(isEncript),
    hostname: hostname,
    port: port,
    path: path
  };
}
```

ref https://github.com/alibaba/anyproxy/pull/450

This changes is patched and this ws proxy work on https://www.websocket.org/echo.html.